File: src/main/java/org/library/Catalogue.java
package org.library;

import java.util.List;
import java.util.ArrayList;
import java.util.HashMap;


public class Catalogue {
    HashMap<String, Book> catalogue;

    public Catalogue(){
        catalogue = new HashMap<>();
    }

    public void addBook(Book book){
        catalogue.put(book.getTitle(), book);
    }

    Book getBook(String title){
        return catalogue.get(title);
    }

    public int getCatalogueSize(){
        return catalogue.size();
    }

    public List<Book> getAllBooks(){
        List<Book> books = new ArrayList<>();
        catalogue.forEach((k, v) -> {
            books.add(v);
        });
        return books;
    }
}

File: main/Catalogue.js
export default class Catalogue {
  constructor() {
    this.catalogue = new Map();
  }

  addBook(book) {
    const title =
      typeof book.getTitle === "function" ? book.getTitle() : book.title;
    this.catalogue.set(title, book);
  }

  getBook(title) {
    return this.catalogue.get(title) || null;
  }

  getCatalogueSize() {
    return this.catalogue.size;
  }

  getAllBooks() {
    return Array.from(this.catalogue.values());
  }
}
